apps/web/src/store/collateral/hooks.ts

```diff
@@ -49,6 +62,25 @@ export function useCollateralChangeCollateralType(): (collateralType: CurrencyKe
   return React.useCallback(
     (collateralType: CurrencyKey) => {
       dispatch(changeCollateralType({ collateralType }));
+      const defaultXChainId = DEFAULT_TOKEN_CHAIN[collateralType];
+      if (defaultXChainId) {
+        dispatch(changeCollateralXChain({ collateralXChain: defaultXChainId }));
+        dispatch(setRecipientNetwork({ recipientNetwork: defaultXChainId }));
+      } else {
+        dispatch(changeCollateralXChain({ collateralXChain: NETWORK_ID === 1 ? '0x1.icon' : '0x2.icon' }));
+        dispatch(setRecipientNetwork({ recipientNetwork: NETWORK_ID === 1 ? '0x1.icon' : '0x2.icon' }));
+      }
+    },
+    [dispatch],
+  );
+}
+
+export function useChangeCollateralXChain(): (collateralXChain: XChainId) => void {
```

Comment:
add the function in `useCollateralActionHandlers` hook

Reply:
moved to https://github.com/balancednetwork/balanced-network-interface/issues/1388